packages/figma-plugin-sync-icons/code.ts

```diff
@@ -0,0 +1,167 @@
+  ) as PageNode | undefined;
+
+  if (!components || !destinationPage) {
+    figma.notify("Eine der Seiten wurde nicht gefunden.");
+    figma.closePlugin();
+    return;
+  }
+  await destinationPage.loadAsync();
+
+  const variables = await figma.variables.getLocalVariablesAsync();
+  const iconColor = variables.find((variable) => variable.name === iconColorVariable);
+  const borderColor = variables.find((variable) => variable.name === borderColorVariable);
+  const backgroundColor = variables.find(
+    (variable: any) => variable.name === backgroundColorVariable,
+  );
+
+  const remoteFrameNames = structure.map((frame: any) => frame.name);
+
+  for (const child of [...(destinationPage.children as any)]) {
+    if (!remoteFrameNames.includes(child.name)) {
+      child.remove();
+      continue;
+    }
+    const remoteFrame = structure.find((frame: any) => frame.name === child.name);
+    const remoteIconNames = remoteFrame?.children.map((icon: any) => icon.name) ?? [];
+
+    for (const icon of [...child.children]) {
+      if (!remoteIconNames.includes(icon.name)) {
+        icon.remove();
+      }
+    }
+  }
+  for (const frame of structure) {
+    const existingFrame = destinationPage.findOne(
+      (node) => node.type === "FRAME" && node.name === frame.name,
+    ) as FrameNode;
+
+    let newFrame = existingFrame;
+
+    if (!existingFrame) {
+      newFrame = figma.createFrame();
+      newFrame.name = frame.name;
+      const { x, y, width, height } = frame.absoluteBoundingBox;
+      newFrame.x = x;
+      newFrame.y = y;
+      newFrame.resize(width, height);
+      newFrame.cornerRadius = 32;
+
+      if (backgroundColor) {
+        newFrame.fills = [
+          {
+            type: "SOLID",
+            color: { r: 0, g: 0, b: 0 },
+            boundVariables: {
+              color: {
+                type: "VARIABLE_ALIAS",
+                id: backgroundColor.id,
+              },
+            },
+          },
+        ];
+      }
+      if (borderColor) {
+        newFrame.strokes = [
+          {
+            type: "SOLID",
+            color: { r: 0, g: 0, b: 0 },
+            boundVariables: {
+              color: {
+                type: "VARIABLE_ALIAS",
+                id: borderColor.id,
+              },
+            },
+          },
+        ];
+        newFrame.strokeWeight = 1;
+      }
+
+      destinationPage.appendChild(newFrame);
+    }
+
+    const existingIconNames = new Set([...newFrame.children].map((child) => child.name));
+    const missingIcons = frame.children.filter((icon: any) => !existingIconNames.has(icon.name));
+
+    const iconPromises = missingIcons.map(async (icon: any) => {
+      const iconKey = components[icon.id].key;
+      if (!iconKey) {
+        return null;
+      }
+
+      try {
+        const component = await figma.importComponentByKeyAsync(iconKey);
+        if (component) {
+          const instance = component.createInstance();
```

Comment:
Afaik we should "disconnect" the copied icons the from original component, so that they are able to override specific parameters without affecting others:
```suggestion
          const instance = component.clone();
```